packages/net/net_address.pony

```diff
@@ -108,18 +108,76 @@ class val NetAddress is Equatable[NetAddress]
       recover String.from_cstring(consume serv) end)
 
   fun eq(that: NetAddress box): Bool =>
-    (this.length == that.length)
-      and (this.family == that.family)
-      and (this.port == that.port)
+      (this._family == that._family)
+      and (this._port == that._port)
       and (host_eq(that))
-      and (this.scope == that.scope)
+      and (this._scope == that._scope)
 
   fun host_eq(that: NetAddress box): Bool =>
     if ip4() then
-      this.addr == that.addr
+      this._addr == that._addr
+    else
+      (this._addr1 == that._addr1)
+        and (this._addr2 == that._addr2)
+        and (this._addr3 == that._addr3)
+        and (this._addr4 == that._addr4)
+    end
+
+  fun length() : U8 =>
+    """
+    For platforms (OSX/FreeBSD) with `length` field as part of 
+    its `struct sockaddr` definition, returns the `length`. 
+    Else (Linux/Windows) returns the size of `sockaddr_in` or `sockaddr_in6`.
+    """
+
+    ifdef linux or windows then
+      (@ponyint_address_length[U32](this)).u8()
     else
-      (this.addr1 == that.addr1)
-        and (this.addr2 == that.addr2)
-        and (this.addr3 == that.addr3)
-        and (this.addr4 == that.addr4)
+      ifdef bigendian then
+        ((_family >> 8) and 0xff).u8()
+      else
+        (_family and 0xff).u8()
+      end
     end
+
+    fun family() : U8 =>
+    """
+      Returns the `family`.
+    """
+
+    ifdef linux or windows then
+      ifdef bigendian then
+        ((_family >> 8) and 0xff).u8()
+      else
+        (_family and 0xff).u8()
+      end
+    else
+      ifdef bigendian then
+        (_family and 0xff).u8()
+      else
+        ((_family >> 8) and 0xff).u8()
+      end
+    end
+
+    fun port() : U16 =>
+      _port
+
+    fun scope() : U32 =>
+      _scope
+
+    fun addr() : (U32 | Array[U32]) =>
```

Comment:
Also each returned integer value (except U8 of course) in any of these methods should be run through a method to convert between network and host byte order. These are `ntohl` for U32 and `ntohs` for U16.